### ISSUE TITLE
False negative | class attribute & global

### ISSUE THREAD

**Issue by mbyrnepr2** [PyCQA/pylint#7098]
there is a somewhat similar behaviour with **global**
````python
A = 1
class Foo:
    global A
    A = 2

print(Foo.A)
````

_Originally posted by @da-dada in https://github.com/PyCQA/pylint/issues/6883#issuecomment-1148987492_

**Comment by mbyrnepr2** [comment]
Created this issue to simplify the scope of https://github.com/PyCQA/pylint/issues/6883.

Running the provided example with Python yields:
```python
Traceback (most recent call last):
  File "example.py", line 6, in <module>
    print(Foo.A)
AttributeError: type object 'Foo' has no attribute 'A'
```
